server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutes.java

```diff
@@ -96,21 +106,35 @@ public void define(Service service) {
         @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Bad request - user param is invalid"),
         @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
     })
-    private TaskIdDto restore(Request request, Response response) {
+    private TaskIdDto restore(Request request, Response response) throws JsonExtractException {
         User userToRestore = extractUser(request);
-        TaskId taskId = taskManager.submit(new DeletedMessagesVaultRestoreTask(userToRestore, vaultRestore));
+        Query query = translate(jsonExtractor.parse(request.body()));
+        TaskId taskId = taskManager.submit(new DeletedMessagesVaultRestoreTask(vaultRestore, userToRestore, query));
         return TaskIdDto.respond(response, taskId);
     }
 
+    private Query translate(QueryDTO queryDTO) {
+        try {
+            return queryTranslator.translate(queryDTO);
+        } catch (QueryTranslator.QueryTranslatorException e) {
+            throw badRequest("Json query passing to the route is invalid", e);
```

Comment:
`Invalid 'query' field` ?

Reply:
I placed whole query in json body, not a field, so, there is no query field

Reply:
true, so you can say "invalid payload"